kernel/src/action_reconciliation/mod.rs

```diff
@@ -0,0 +1,289 @@
+//! # This module implements APIs related to action reconciliation.
```

Comment:
nit: this will render as a pretty big heading so perhaps consider something like

```suggestion
//! # Action Reconciliation
//! This module implements APIs related to action reconciliation.
```